src/Database/Statement/Statement.php

```diff
@@ -227,12 +227,9 @@ protected function convertMode(string|int $mode): int
             return $mode;
         }
 
-        $mode = static::MODE_NAME_MAP[$mode] ?? null;
-        if ($mode !== null) {
```

Comment:
I would have probably just turned this around to bail early and have the last line do the happy case return.

Reply:
In majority cases the mode would be correct so better to have that first.

Reply:
Hm, I dunno.
The bail early principle is used throughout the code for consistency and to reduce cognitive load (easier human review).
I guess here its fine, since it is simple enough and almost a one liner.